### DISCUSSION TITLE
Add Expo config plugin for messaging module [iOS]

### DISCUSSION THREAD

**Discussion by bazter** [invertase/react-native-firebase#6998] [Ideas]
Hello, I'd like to request an expo config plugin, that would modify the _AppDelegate.mm_ file according to [docs](https://rnfirebase.io/messaging/usage#background-application-state).

The change consists of two lines I have to manually add every time I run `expo prebuild --clean`:

```
//AppDelegate.mm

// top of the file
#import "RNFBMessagingModule.h"

// inside the body of didFinishLaunchingWithOptions
self.initialProps = [RNFBMessagingModule addCustomPropsToUserProps:nil withLaunchOptions:launchOptions];
```

This change is needed if we'd like to skip unnecessary app bootstrap during background notifications on iOS.
The provided snippet is from latest RN (0.71.x) and it was actually a bit different before, so some backward compatibility might be needed for older react-native versions.

The plugin could be similar to the one from the _app_ module, which does very similar modification. For reference, please see `@react-native-firebase/app/plugin/src/ios/appDelegate.ts`

**Comment by mikehardy** [comment]
Expo plugins are community supported, if you propose a PR implementing a new plugin, we can merge it.
